typescript/src/chain.ts

```diff
@@ -92,16 +105,39 @@ export interface Bridge {
 
   /**
    * Gets a pending redemption from the on-chain contract.
-   * @param walletPubKeyHash The wallet public key hash that identifies the
-   *        pending redemption (along with the redeemer output script). Must be
-   *        unprefixed
-   * @param redeemerOutputScript The redeemer output script that identifies the
-   *        pending redemption (along with the wallet public key hash). Must be
-   *        un-prefixed and not prepended with length
+   * @param walletPublicKey Bitcoin public key of the wallet the request is
+   *        targeted to. Must be in the compressed form (33 bytes long with 02
+   *        or 03 prefix).
+   * @param redeemerOutputScript The redeemer output script the redeemed funds
+   *        are supposed to be locked on. Must be un-prefixed and not prepended
+   *        with length.
    * @returns Promise with the pending redemption.
    */
   pendingRedemptions(
-    walletPubKeyHash: string,
+    walletPublicKey: string,
+    redeemerOutputScript: string
+  ): Promise<RedemptionRequest>
+
+  /**
+   * Gets a timed-out redemption from the on-chain contract.
+   * @param walletPublicKey Bitcoin public key of the wallet the request is
+   *        targeted to. Must be in the compressed form (33 bytes long with 02
+   *        or 03 prefix).
+   * @param redeemerOutputScript The redeemer output script the redeemed funds
+   *        are supposed to be locked on. Must be un-prefixed and not prepended
+   *        with length.
+   * @returns Promise with the pending redemption.
+   */
+  timedOutRedemptions(
```

Comment:
Shouldn't this be called `timedOutRedemption` since it return only one redemption request based on the `redemptionKey`? Or maybe I don't understand something 🤔

Reply:
This name is chosen deliberately to match the name of the function exposed by the `Bridge` function. That should make the lib <-> contract integration easier to understand.

Reply:
I understand, but I was just wondering if the function name really reflects what the function does.

https://github.com/keep-network/tbtc-v2/blob/21eaf4624d160a0d0810f1579b8ac5baf27e9b44/solidity/contracts/bridge/Bridge.sol#L1466-L1472

Maybe we should consider changing it to `timedOutRedemption` in solidity contract (and then do the sime in ts lib), since, like i said, it looks to me that it only returns one redemption request.

Reply:
This is a Solidity convention, we use the same getter name as the object the getter refers to. I don't think we can change it now because the contract codebase is frozen now.